### PR TITLE
Expose managehome parameter to user

### PR DESCRIPTION
**Pull request by bbriggs**
Managehome is set to true in local_user module, which is a good default. However, I would like to keep the default but expose the variable to the user.

### CODE REVIEW

**Comment by bbriggs** [comment]
Looks like some of the older checks are complaining about str2bool. I'll take a look and run this in my own TravisCI so that it doesn't pollute your inbox. Look for an update tonight or Monday.

File: manifests/init.pp [hunk 1]
```diff
@@ -35,6 +35,7 @@
   $password_max_age    = 90,
   $shell               = '/bin/bash',
   $home                = "/home/${name}",
+  $managehome          = true,
```
Comment (rnelson0) [inline manifests/init.pp]: This defaults to false in the user resource, so should here as well.

File: manifests/init.pp [hunk 2]
```diff
@@ -50,6 +51,7 @@
     validate_array($groups)
     validate_string($password)
     validate_integer($last_change)
+    str2bool($managehome)
```
Comment (rnelson0) [inline manifests/init.pp]: I think you want `validate_bool()`?

File: manifests/init.pp (hunk 1 above)
Comment (bbriggs) [inline manifests/init.pp]: It originally hard-coded to true in your manifest, so I was matching
that. You're right about it defaulting to false in the built-in type,
though. So set it to false then?

**Comment by bbriggs** [comment]
Using validate_bool will work as long as the user passes something like [`true` instead of `"true"`](https://forge.puppetlabs.com/puppetlabs/stdlib#usage). Is that a behavior we want to keep? It might cause frustration since people can normally pass bools in as strings thanks to people accounting for dangerous type casting on input. 

IMO, we should use validate_bool and note in the docs that this is a bool and should not be a string. This most recent commit reflects that change.

**Comment by rnelson0** [comment]
Awesome, love it, thanks! Can you squash it to a single comment?

**Comment by bbriggs** [comment]
Sure. I don't know of a way to squash an existing branch, so I'll just send over a new PR from a different branch.

**Comment by rnelson0** [comment]
`git rebase -i origin/master` is a good method to squash commits.

**Comment by bbriggs** [comment]
I should have done Commitmas!

**Pull request closed without merging**